src/preprocess/preprocess_vhi.py

```diff
@@ -0,0 +1,192 @@
+import time
+import xarray as xr
+import pandas as pd
+import numpy as np
+from typing import Optional, Tuple, Any
+
+from xarray import Dataset, DataArray
+from pandas._libs.tslibs.timestamps import Timestamp
+
+# ------------------------------------------------------------------------------
+# 1. assign time stamp
+# ------------------------------------------------------------------------------
+
+
+def extract_timestamp(ds: Dataset,
```

Comment:
What is the motivation for keeping these functions in a seperate file? Can they be part of the `VHIPreprocessor` class?